src/temporal/t.rast.boxplot/t.rast.boxplot.py

```diff
@@ -1,631 +1,802 @@
+        else:
+            date_fmt = mdates.DateFormatter("%Y")
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+    if "month" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("%Y-%m")
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+    if "day" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("%Y-%m-%d")
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+    if "hour" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("hour %H")
+        start_time = mdates.date2num(rast_dates[0]) - 1 / 48 * temp_lngt
+        end_time = mdates.date2num(rast_dates[-1]) + 1 / 48 * temp_lngt
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+            ax.set_xlim(mdates.num2date(start_time), mdates.num2date(end_time))
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+            ax.set_ylim(mdates.num2date(start_time), mdates.num2date(end_time))
+    if "minute" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("min. %M")
+        start_time = mdates.date2num(rast_dates[0]) - 1 / 2880 * temp_lngt
+        end_time = mdates.date2num(rast_dates[-1]) + 1 / 2880 * temp_lngt
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+            ax.set_xlim(mdates.num2date(start_time), mdates.num2date(end_time))
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+            ax.set_ylim(mdates.num2date(start_time), mdates.num2date(end_time))
+    if "second" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("sec. %S")
+        start_time = mdates.date2num(rast_dates[0]) - 1 / 172800 * temp_lngt
+        end_time = mdates.date2num(rast_dates[-1]) + 1 / 172800 * temp_lngt
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+            ax.set_xlim(mdates.num2date(start_time), mdates.num2date(end_time))
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+            ax.set_ylim(mdates.num2date(start_time), mdates.num2date(end_time))
+    return ax
+
+
+def main(options, flags):
+    """
+    Draws the boxplot of raster values. Optionally, this is done per
+    category of a zonal raster layer
+    """
+
+    global matplotlib
```

Comment:
This could be a standalone function:

```python
def lazy_import_py_modules():
    """Lazy import Py modules"""
    global matplotlib
    global plt

    # lazy import matplotlib
    try:
        import matplotlib

        matplotlib.use("WXAgg")
        from matplotlib import pyplot as plt
    except ModuleNotFoundError:
        gs.fatal(_("Matplotlib is not installed. Please, install it."))
```